core/src/main/scala/akka/kafka/internal/CommitCollectorStage.scala

```diff
@@ -46,79 +46,77 @@ private final class CommitCollectorStageLogic(
 
   override protected def logSource: Class[_] = classOf[CommitCollectorStageLogic]
 
-  // ---- initialization
+  private var emitOnPull = false
```

Comment:
Naming suggestion: `pushOnNextPull`.  We only emit now on shutdown and the variable is toggled on the next pull.